src/main/java/uk/gov/ons/census/fwmt/csvservice/implementation/postcodeloader/RejectionProcessor.java

```diff
@@ -0,0 +1,116 @@
+package uk.gov.ons.census.fwmt.csvservice.implementation.postcodeloader;
+
+import com.opencsv.bean.ColumnPositionMappingStrategy;
+import com.opencsv.bean.StatefulBeanToCsv;
+import com.opencsv.bean.StatefulBeanToCsvBuilder;
+import com.opencsv.exceptions.CsvDataTypeMismatchException;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.fwmt.common.error.GatewayException;
+import uk.gov.ons.census.fwmt.csvservice.dto.AddressCheckListing;
+import uk.gov.ons.census.fwmt.csvservice.dto.RejectionReport;
+import uk.gov.ons.census.fwmt.csvservice.utils.CsvServiceUtils;
+import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+import static uk.gov.ons.census.fwmt.csvservice.implementation.postcodeloader.LookupFileLoaderEventsConfig.CREATED_REJECTION_FILE;
+
+@Slf4j
+@Component
+public class RejectionProcessor {
+
+  @Value("${gcpBucket.rejectLocation}")
+  String location;
+  @Autowired
+  private CsvServiceUtils csvServiceUtils;
+  @Autowired
+  private GatewayEventManager gatewayEventManager;
+
+  public void createErrorReports(List<AddressCheckListing> rejectionsList,
+      List<RejectionReport> rejectedReportList) throws GatewayException {
+    String timeStamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH-mm-ss"));
+    createRejectionCsv(rejectionsList, timeStamp);
+    createRejectionReport(rejectedReportList,timeStamp);
+  }
+
+  private void createRejectionReport(
+      List<RejectionReport> rejectedReportList, String timeStamp) throws GatewayException {
+    File file;
+    try {
+      file = File.createTempFile("rejectionTemp", ".report");
+    } catch (IOException e) {
+      throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, e, "Failed creating temp file to write to.");
+    }
+
+    try (Writer writer = new FileWriter(file.getAbsolutePath(), StandardCharsets.UTF_8);
+    InputStream inputStream = new FileInputStream(file)){
+
+      for (RejectionReport rejectionReport: rejectedReportList) {
+        try {
+          writer.write("Case Reference: " + rejectionReport.getCaseRef() + " | " + rejectionReport.getReason());
+          writer.write("\n");
+        } catch (IOException e) {
+          throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, e, "Failed creating temp file to write to.");
+        }
+      }
+
+      String filename = "reject_" + timeStamp + ".txt";
```

Comment:
Put this and the following lines after the catch. The reason being the clean up and flush are done in an invisible finally (probably), so if the flush isnt called before the u call the uploadFile function, u might upload an uncomplete file